core/object/object.cpp

```diff
@@ -41,6 +41,7 @@
 #include "core/string/translation.h"
 #include "core/templates/local_vector.h"
 #include "core/variant/typed_array.h"
+#include "scene/main/node.h"
```

Comment:
Core should really not have dependencies in `scene` I'd say, I think things should be local to `Node`

Reply:
I need it to check whether the object is a Node. Is there a way of doing this would including `node.h`?

Reply:
To check for type you should be able to use `is_class("Node")`

Reply:
This works, but I need to call `get_debug_path()` which only exists in Node. I can't use `call()` since I can't call a non-`const` method within a `const` method (`get_debug_path()` is `const`, `call()` is not).

I could use `call_const()` which seems to do what I want, but I don't know how to call it without any arguments since all arguments are required.

I've tried this, but it doesn't compile due to the Callable::CallError argument (I just want to discard it, I don't need to read it):

```cpp
call_const(SNAME("get_debug_path"), (const Variant **)nullptr, 0, (Callable::CallError){}), p_signal, get_class()
```

Reply:
Could use a virtual method instead I'd say, but might be worth it to break the isolation here